lib/actions/user.js

```diff
@@ -181,3 +188,91 @@ export function deleteUserMonitoredTrip (id) {
+      //   and cache the previous value and verification state so they can be reverted.
+      //   The full user record will be updated upon the user clicking Finish/Save preferences.
+      // TODO: Figure out what should happen if the user refreshes the browser at this stage.
+
+      // Make a clone of the original userData object.
+      const newUserData = clone(originalUserData)
+      const previousPhoneNumber = newUserData.phoneNumber
+      const previousIsPhoneNumberVerified = newUserData.isPhoneNumberVerified
+
+      newUserData.phoneNumber = newPhoneNumber
+      newUserData.isPhoneNumberVerified = false
+
+      const userUpdateResult = await updateUser(otpMiddleware, accessToken, newUserData)
+
+      if (userUpdateResult.status === 'success' && userUpdateResult.data) {
+        const sendSmsResult = await sendPhoneVerificationSms(otpMiddleware, accessToken, newUserData.id)
+        if (sendSmsResult.status === 'success') {
+          // Update application state on success.
+          dispatch(setCurrentUser({ accessToken, user: userUpdateResult.data }))
+        } else {
+          alert(`An error was encountered:\n${JSON.stringify(sendSmsResult)}`)
+
+          // FIXME: Also if there was an error in sending the verificaton request,
+          // revert the phone number and verification status in the database.
+          newUserData.phoneNumber = previousPhoneNumber
+          newUserData.isPhoneNumberVerified = previousIsPhoneNumberVerified
+          await updateUser(otpMiddleware, accessToken, newUserData)
+        }
+      } else {
+        alert(`An error was encountered:\n${JSON.stringify(userUpdateResult)}`)
+      }
+    }
+  }
+}
+
+/**
+ * Validate the phone number verification code for the logged-in user.
+ */
+export function verifyPhoneNumber (originalUserData, code) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken } = user
+      const sendResult = await validatePhoneVerificationCode(otpMiddleware, accessToken, originalUserData.id, code)
+      if (sendResult.status === 'success' && sendResult.data) {
+        if (sendResult.data.status === 'approved') {
+          // Make a clone of the original userData object.
+          const newUserData = clone(originalUserData)
+
+          // Update phone number and verification in database record.
+          // FIXME: The call to updateUer below assumes the middleware requires saving the user's phone number prior to verification.
+          newUserData.isPhoneNumberVerified = true
+          newUserData.notificationChannel = 'sms'
+          const userUpdateResult = await updateUser(otpMiddleware, accessToken, newUserData)
+
+          if (userUpdateResult.status === 'success' && userUpdateResult.data) {
+            alert('Your phone is now verified and set to receive trip notifications.')
```

Comment:
We need to stop embedding alerts in the code...

Reply:
I'll remove this one, as there is already a check mark feedback shown when the number is verified.